src/AutoRest.CSharp/Mgmt/Decorator/ParameterMappingBuilder.cs

```diff
@@ -200,16 +200,21 @@ private static IEnumerable<List<Segment>> SplitDiffIntoPairs(RequestPath diff)
 
         public static FormattableString GetValueExpression(CSharpType type, FormattableString rawExpression)
         {
-            if (TypeFactory.IsStringLike(type))
+            // if the type is string
+            if (type.EqualsIgnoreNullable(typeof(string)))
+                return rawExpression;
+
+            // or if the type is extensible enum and its underlying type is also string
+            if (type is { IsFrameworkType: false, Implementation: EnumType { IsExtensible: true, ValueType: { } enumValueType } } && enumValueType.EqualsIgnoreNullable(typeof(string)))
```

Comment:
how about having a private method to check this and the above condition?

Reply:
well since we only use this once, I do not think it is worthy to write a method for it